### DISCUSSION TITLE
Garage ceiling project

### DISCUSSION THREAD

**Discussion by waynen1967** [PlummersSoftwareLLC/NightDriverStrip#653] [General]
Hello, wanting to do a fairly large, 34 square foot ceiling with hexagon RGB leds. What I have already has each 3 leds can be controlled, I can keep buying these on make something on my own or buy what's needed, but wanting to know where to start as far as which ESP32 would be best, an idea of how many it would take and how many places I may need to inject power to keep the colors accurate and bright. Looking at mounting them into a breakout or breadboard with screw terminals and using 3d printed cases, all the wiring can easily be shoved above the 2x2 drop ceiling. Can someone help me get started in the right direction? Thanks in advance.
Wayne

**Comment by davepl** [comment]
You can run about 1000 LEDs on each channel if you still want 30fps.  So 8 channels of 1000 each, for a total of 8000.  If doing WiFi, I recommend something with PSRAM to buffer that many channels.

You can do more (like twice as many) at half the refresh rate, but the fundamental data rate limits how many you can feed.

Cheer,
Dave

**Reply by waynen1967** [reply]
Thanks, I think I'll need less than 4,000 total pixels, and that's if I am able to replace these LED's with indivisibly addressable ones, if I just buy these over the counter and let them controlling every 3 LED's then far less. Would one ESP32 be able to reach all around that 34 square feet with data ok then? I'm not sure how to even say what I want here, will try. So if I put the data in on one end of a row of these hexagons, it will control to the end of the line, then if the next row also gets data, then that row completely duplicates the first row, and that may be god enough on it's own, but how would I also be able to control the other axis? Somehow I tell the software where the rows are and it does the rest? Hope these aren't dumb questions, I am new to this completely, but have done literally millions of RGB strips under DMX control so hoping I'm able to get all this. I've heard the term pixel mapping, but don't understand well enough to know if this software is capable of doing what I think I want. But again, if all 8 rows can just do the same thing, it will still be an amazing effect, so I may just settle for that. Thank you for the quick response!
Wayne

**Comment by davepl** [comment]
Each LED regenerates the signal, so you will likely have to inject power every 200 or so LEDs (at 5V) or every 1000 (at 12v).  But the data gets regenerated by each LED.  I don’t know about a long run of data wire in front of it, I’ve never done more than about 10 feet.  But if you make it to first LED, it’ll be fine from there.
- Dave

**Comment by robertlipe** [comment]
On eggs and baskets.. remember that ws2811/2 are a daisy chain/bucket
brigade. If pixel x in a chain fails, all pixels from x to the end quit.
Failure rate is governed largely by cooling/brightness and just the
statistics of 4000leds all working. It's a reason to prefer shorter chains
to longer ones...and to plan for serviceability.

Granted, a garage light probably isn't on super bright around the clock, so
statistics may be on your side.


Good luck

**Comment by robertlipe** [comment]
10-15 feet of good, enclosed wire is about the threshold of the 3.3v if the
esp32 because the threshold of the 281x isn't a whole lot before that.

A sacrificial pixel (one pixel near the board) will click in the signal and
retransmit at 5v for a little pick-me-up. Level shifters, sometimes called
retransmitters, for multiple channels are readily available and
inexpensive. Honestly, in most cases, it's easier for power management to
get your esp32 near 8 strips than it is to add shifters and 8 wires in most
cases.

Remember that your esp and your power rails need to share a ground.

**Comment by waynen1967** [comment]
I'm still lost as far as what exact ESP32 I actually need. I see on the installation page it shows a lot of different ones, and once selected, you can see another list of what projects they will do, at least I think that's what it is. I ordered these, https://shorturl.at/Ixs2t wanting the screw terminals to make my connections, and also these, https://shorturl.at/zzfFf but when I got them, the breakout board has 38 pins, and the ESP's have 44... Would these work on the breakout boards I got and be able to install the nightdriverled firmware? https://shorturl.at/7dET3

**Comment by robertlipe** [comment]
I use the 44 pin boards heavily. It's been only recently that I could find
44 pin BOBs. The good news is that for simple configurations you can dangle
the ends off the common 38 boards and ignore the hanging pins. Just pick
your pinouts carefully.

As this is largely a DIY project and you're doing something new, there
won't be an exact BOM and instructble. You have latitude on the precise
board you use, the targets you start with to customize, etc.  Either of the
boards you list are reasonable. I like the n16r8s as they're fast, have
flexible. Pin mappings, and have large, fast(ish) ram. That's what I use
them, though they ARE overkill for lots of things and they're a large if
all you really need is a  nano-class board. I just prefer to inventory and
standardize my projects around a single $4 part instead of wondering if I
can squeeze it into a $3 part. But those are your decisions with a diy
project.

Good luck
